src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php

```diff
@@ -33,6 +34,18 @@ public function getConfigTreeBuilder(): TreeBuilder
 
         $rootNode
             ->docUrl('https://symfony.com/doc/{version:major}.{version:minor}/reference/configuration/twig.html', 'symfony/twig-bundle')
+            ->beforeNormalization()
+            ->ifTrue(fn ($v) => \is_array($v) && \array_key_exists('exception_controller', $v))
```

Comment:
using ifTrue implicitly means the node can be cast from bool when generating array shapes/fluent php
```suggestion
            ->ifArray()
```

Reply:
this looks like a shortcoming of the generator when using `ifTrue` with a callback though.

Reply:
it's like that before 4.4 🤷 nobody cared until we worked on shapes.

Reply:
No. When passing a callable, we don't support boolean input. We support input for which the closure returns `true`, which could be something else.

Reply:
Right, I checked the code again, this turns the type into "mixed", since anything could be accepted after that call.